### PR TITLE
New grafana columns

### PR DESCRIPTION
**Pull request by markschnitziusgovuk**
Rearrangement of the columns in the Grafana table, including some new columns.

image

### CODE REVIEW

File: cloudformation/calculation-athena.yaml [hunk 1]
```diff
@@ -193,31 +193,25 @@ Resources:
...
-                WHEN (txn.price IS NULL) THEN ${QueryMessageRateCardDataMissing}
-                ELSE (CASE WHEN (COALESCE(txn.price, 0) = 0) THEN (CASE WHEN (COALESCE(invoice.price, 0) = 0) THEN 0 ELSE null END) ELSE ((100 * (COALESCE(invoice.price, 0) - txn.price)) / txn.price) END) END) Difference
+            , "concat"(U&'\00A3', "regexp_replace"(CAST(CAST(invoice.price AS decimal(10,2)) AS varchar), '(\d)(?=(\d{3})+\.)', '$1,')) billing_price_formatted
+            , "concat"(U&'\00A3', "regexp_replace"(CAST(CAST(txn.price AS decimal(10,2)) AS varchar), '(\d)(?=(\d{3})+\.)', '$1,')) transaction_price_formatted
+            , (CASE WHEN ((invoice.price = 0) AND (txn.price = 0)) THEN -1.23456701E6 WHEN ((invoice.price IS NULL) AND (invoice.quantity IS NULL)) THEN -1.23456703E6 WHEN ((txn.price IS NULL) AND (txn.quantity IS NULL)) THEN -1.23456704E6 WHEN (txn.price IS NULL) THEN -1.23456702E6 ELSE (CASE WHEN (COALESCE(txn.price, 0) = 0) THEN (CASE WHEN (COALESCE(invoice.price, 0) = 0) THEN 0 ELSE null END) ELSE ((100 * (COALESCE(invoice.price, 0) - txn.price)) / txn.price) END) END) price_difference_percentage
```
[the remark this answers is not in the document]
Comment (markschnitziusgovuk) [reply cloudformation/calculation-athena.yaml:205]: The parameters and formatting are mistakes, thanks -- I was modifying the queries in AWS and then copying/pasting them back into the file here.  Will fix.

File: cloudformation/calculation-athena.yaml [hunk 2]
```diff
@@ -193,31 +193,25 @@ Resources:
         Database: !Ref CalculationsDB
         Name: btm_billing_and_transactions_curated
         Query: !Sub |
+          CREATE OR REPLACE VIEW "btm_billing_and_transactions_curated" AS
```
[the remark this answers is not in the document]
Comment (markschnitziusgovuk) [reply cloudformation/calculation-athena.yaml:196]: Yeah, copy-paste error and failed to test deployment...  Fixing.

File: grafana-json/billing-reconciliation-dashboard.preprod.json
Comment (MelAtGDS) [inline grafana-json/billing-reconciliation-dashboard.preprod.json:1]: Filename on line 50 is `raw-Invoice-validFile.pdf`, but line 60 adds another .pdf at the end of the file name.

File: integration_tests/tests/billing-and-transaction-view-tests.ts
Comment (MelAtGDS) [inline integration_tests/tests/billing-and-transaction-view-tests.ts:1]: Filename on line 50 is raw-Invoice-validFile.pdf, but line 60 adds another .pdf at the end of the file name.

Comment (markschnitziusgovuk) [reply integration_tests/tests/billing-and-transaction-view-tests.ts:1]: Fixing.

**Review by MelAtGDS — APPROVED** [review]

**Pull request merged**